### PR TITLE
CUT-2378: -Attribues Object needs custom type

### PR DESCRIPTION
**Pull request by gweinjc**
## Issues
* [CUT-2378](https://jumpcloud.atlassian.net/browse/CUT-2378) - -Attribues Object needs custom type

## What does this solve?
Through some discovery it was determined that the -Attributes object is being flattened by the SDKs and does not necessarily need a custom type. 

As an example the following script would set sudo to enabled and withoutPassword to false
```powershell
$attributes = [JumpCloud.SDK.V2.Models.IGraphOperationSystemAttributes]@{ 'sudo' = @{'enabled' = $true; 'withoutPassword' = $false } }
Set-JcSdkSystemAssociation -SystemId 61982961719e942987898f39 -id 5ef5fe69a899bd46963ab24a -op 'update' -type 'user' -Attributes $attributes
```

We can achieve the same result by doing the following:
```powershell
Set-JcSdkSystemAssociation -SystemId 61982961719e942987898f39 -id 5ef5fe69a899bd46963ab24a -op 'update' -type 'user' -Attributes @{ 'sudoEnabled' = $true; 'SudoWithoutPassword' = $true }
```

This is due to the SDKs doing a sort of flattening to the sudo object and allowing us to instead call the sudoEnabled or SudoWithoutPassword property to set those values.

Rather than creating a custom object, we can update the documentation for this functionality

This behavior can also be applied to the `Set-JcSdkUserAssociation` function

## Is there anything particularly tricky?
N/A

## How should this be tested?

Pull the branch and import the updated version of the V2 SDK
Attempt to run the following script and ensure that the results are updated in the console:
```powershell
Set-JcSdkSystemAssociation -SystemId <SystemID> -id <UserID> -op 'update' -type 'user' -Attributes @{ 'sudoEnabled' = $true; 'SudoWithoutPassword' = $true }
```

[CUT-2378]: https://jumpcloud.atlassian.net/browse/CUT-2378?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: SDKs/PowerShell/JumpCloud.SDK.V2/custom/generated/Set-JcSdkSystemAssociation.ps1 [hunk 1]
```diff
@@ -48,13 +48,22 @@ curl -X POST https://console.jumpcloud.com/api/v2/systems/{System_ID}/associatio
...
+PS C:\> Set-JcSdkSystemAssociation -SystemId <SystemID> -id <UserID> -op 'add' -type 'user' -Attributes @{ 'sudoEnabled' = $true; 'SudoWithoutPassword' = $true }
 
-{{ Add output here }}
 .Example
-PS C:\> {{ Add code here }}
+PS C:\> Set-JcSdkSystemAssociation -SystemId <SystemID> -id <UserID> -op 'update' -type 'user' -Attributes @{ 'sudoEnabled' = $true; 'SudoWithoutPassword' = $true }
 
-{{ Add output here }}
+.Example
+PS C:\> Set-JcSdkSystemAssociation -SystemId <SystemID> -id <UserID> -op 'remove' -type 'user' -Attributes @{ 'sudoEnabled' = $false; 'SudoWithoutPassword' = $false }
```
Comment (jworkmanjc) [inline SDKs/PowerShell/JumpCloud.SDK.V2/custom/generated/Set-JcSdkSystemAssociation.ps1:57]: One thing some customers might like better is just. It does the same thing, that's the whole bit we added to the build custom functions

`Set-JcSdkSystemAssociation -SystemId <SystemID> -id <UserID> -op 'remove' -type 'user'`

File: SDKs/PowerShell/JumpCloud.SDK.V2/custom/generated/Set-JcSdkSystemAssociation.ps1 [hunk 2]
```diff
@@ -48,13 +48,22 @@ curl -X POST https://console.jumpcloud.com/api/v2/systems/{System_ID}/associatio
...
+
+.Example
+PS C:\> Set-JcSdkSystemAssociation -SystemId <SystemID> -id <UserGroupID> -op 'add' -type 'user_group' -Attributes @{ 'sudoEnabled' = $true; 'SudoWithoutPassword' = $false }
+
+.Example
+PS C:\> Set-JcSdkSystemAssociation -SystemId <SystemID> -id <UserGroupID> -op 'update' -type 'user_group' -Attributes @{ 'sudoEnabled' = $true; 'SudoWithoutPassword' = $false }
+
+.Example
```
Comment (jworkmanjc) [inline SDKs/PowerShell/JumpCloud.SDK.V2/custom/generated/Set-JcSdkSystemAssociation.ps1:65]: same thing here, there's no need to pass in attributes when you remove a system association we can just do:

`Set-JcSdkSystemAssociation -SystemId <SystemID> -id <UserGroupID> -op 'remove' -type 'user_group'`

File: SDKs/PowerShell/JumpCloud.SDK.V2/custom/generated/Set-JcSdkUserAssociation.ps1 [hunk 3]
```diff
@@ -48,13 +48,23 @@ curl -X POST https://console.jumpcloud.com/api/v2/users/{UserID}/associations \\
...
-PS C:\> {{ Add code here }}
+PS C:\> Set-JcSdkUserAssociation -UserId <UserID> -id <SystemID> -op 'add' -type 'user' -Attributes @{ 'sudoEnabled' = $true; 'SudoWithoutPassword' = $true }
 
-{{ Add output here }}
 .Example
-PS C:\> {{ Add code here }}
+PS C:\> Set-JcSdkUserAssociation -UserId <UserID> -id <SystemID> -op 'update' -type 'user' -Attributes @{ 'sudoEnabled' = $true; 'SudoWithoutPassword' = $true }
+
+.Example
+PS C:\> Set-JcSdkUserAssociation -UserId <UserID> -id <SystemID> -op 'remove' -type 'user' -Attributes @{ 'sudoEnabled' = $false; 'SudoWithoutPassword' = $false }
```
Comment (jworkmanjc) [inline SDKs/PowerShell/JumpCloud.SDK.V2/custom/generated/Set-JcSdkUserAssociation.ps1:57]:
```suggestion
PS C:\> Set-JcSdkUserAssociation -UserId <UserID> -id <SystemID> -op 'remove' -type 'user'
```

File: SDKs/PowerShell/JumpCloud.SDK.V2/custom/generated/Set-JcSdkUserAssociation.ps1 [hunk 4]
```diff
@@ -48,13 +48,23 @@ curl -X POST https://console.jumpcloud.com/api/v2/users/{UserID}/associations \\
...
+
+.Example
+PS C:\> Set-JcSdkUserAssociation -UserId <UserID> -id <SystemGroupID> -op 'add' -type 'system_group' -Attributes @{ 'sudoEnabled' = $true; 'SudoWithoutPassword' = $false }
+
+.Example
+PS C:\> Set-JcSdkUserAssociation -UserId <UserID> -id <SystemGroupID> -op 'update' -type 'system_group' -Attributes @{ 'sudoEnabled' = $true; 'SudoWithoutPassword' = $false }
+
+.Example
+PS C:\> Set-JcSdkUserAssociation -UserId <UserID> -id <SystemGroupID> -op 'remove' -type 'system_group' -Attributes @{ 'sudoEnabled' = $false; 'SudoWithoutPassword' = $false }
```
Comment (jworkmanjc) [inline SDKs/PowerShell/JumpCloud.SDK.V2/custom/generated/Set-JcSdkUserAssociation.ps1:66]:
```suggestion
PS C:\> Set-JcSdkUserAssociation -UserId <UserID> -id <SystemGroupID> -op 'remove' -type 'system_group'
```

**Review by jworkmanjc — CHANGES_REQUESTED** [review]
I think everything looks fine to me, Just two suggestions on the examples, if you remove associations from on object you technically don't need to pass in an attributes object. I think we should mark our examples this way but open for debate

**Review by jworkmanjc — APPROVED** [review]
Thanks for the examples!

**Review by kmaranionjc — APPROVED** [review]
Tests worked and code looks good. Great work

**Pull request merged**